src/app/api/delegate/platform/route.ts

```diff
@@ -70,23 +72,32 @@ export async function POST(req: Request) {
 
     console.log('tx hash', txHash);
 
+    // Wait for transaction to be mined
+    const receipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+    console.log('Mint Transaction Status', receipt.status);
+
     await publicClient.waitForTransactionReceipt({
-      confirmations: 1,
+      confirmations: 2,
       hash: txHash,
     });
 
-    const id = await publicClient.readContract({
-      address: config.contracts.talentLayerPlatformId,
-      abi: TalentLayerPlatformID.abi,
-      functionName: 'ids',
-      args: [body.address],
-    });
+    let id: bigint = 0n;
+    let retries = 0;
+
+    while (id === 0n && retries < MAX_RETRIES) {
+      id = (await publicClient.readContract({
```

Comment:
move here: 
 await publicClient.waitForTransactionReceipt({
      confirmations: 1,
      confirmations: 2,
      hash: txHash,
    });

and every time you loop you wait one confirmations more